coalib/results/Diff.py

```diff
@@ -351,14 +351,20 @@ def __bool__(self):
     def delete_line(self, line_nr):
         """
         Mark the given line nr as deleted. The first line is line number 1.
+
+        Raises an exception if line nr doesn't exist in the diff.
         """
+        line_nr = 0 if (len(self._file) < line_nr) else line_nr
```

Comment:
Sorry, no parens around the if condition (unrelated to what @meetmangukiya said though)